_posts/2018-02-16-Java-Time2.md

```diff
@@ -0,0 +1,1564 @@
+        Zone zone = null;
+
+        try {
+            zone = new Zone(); // 7
+            zone.setName(zoneName); // 8
+            domain.createChild(zone); // 9
+        } catch (RestException e) {
+            e.printStackTrace();
+        }
+        
+        return zone;
+    }
+    
+    public List<Zone> fetchAllZonesForDomain(Domain domain) {
+        ZonesFetcher fetcher = domain.getZones(); // 10
+        List<Zone> list = new ArrayList<>();
+
+        try {
+            if (fetcher.count() > 0) { // 11
+                list = fetcher.get(); // 12
+            }
+        } catch (RestException e) {
+            e.printStackTrace();
+        }
+
+        return list;
+    }
+    
+    public void deleteZone(Zone zone) { // 13
+        if (zone == null) {
+            System.out.println("Zone does not exist");
+        } else {
+            System.out.println("Zone " + zone.getName() + " will be deleted");
+            try {
+                zone.delete(); // 14
+            } catch (RestException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+}
+{% endhighlight %}
+
+In Line 1 of the `verifyZone` method, we supply arguments for the parent Domain object and the Zone name.
+
+Line 2 checks whether an existing Zone object matches the name that is passed in via `NetworkDescriptor`. If the Zone already exists, then we return that Zone object.  If the Zone does not exist, then in Line 3 we call the `createZone` method to create and return a new Zone object.
+
+Line 4 of the `fetchZone` method sets up a search filter based on the name of our zone.  In Line 5, we create an instance of `ZonesFetcher`.  Then Line 6 retrieves the Zone object matching the name in the search filter. 
+
+In Line 7 of the `createZone` method, we call the constructor of the VSPK Zone class to create a brand new Zone object and then in Line 8 we assign our zone name.  Finally, in Line 9, we attach our new Zone object to the already existing Domain object.
+
+Lines 10 to 14 of methods `fetchAllZonesForDomain` and `deleteZone` will be covered later when we discuss how to remove our network layout from the VSD.
+
+# Adding the Zone to our Network
+
+Here is the updated version of the `NetworkBuilder` class that integrates with `NetworkZoneHandler`.  There are 2 minor changes as shown below.  At Line 1, we instantiate the new handler and at Line 2, we call its `verifyZone` method and pass it our existing Domain object and the `NetworkDescriptor` object which contains the name of our zone.
+
+{% highlight java %}
+import java.util.List;
+import net.nuagenetworks.bambou.RestException;
+import net.nuagenetworks.vspk.v5_0.Zone;
+import net.nuagenetworks.vspk.v5_0.Domain;
+import net.nuagenetworks.vspk.v5_0.DomainTemplate;
+import net.nuagenetworks.vspk.v5_0.Enterprise;
+import net.nuagenetworks.vspk.v5_0.VSDSession;
+
+public class NetworkBuilder {
+    private String url = "https://135.228.4.108:8443";

```

Comment:
Don't use a 135.... IP in documentation, use one from https://tools.ietf.org/html/rfc5737

Suggestion: `198.51.100.108`